### PR TITLE
Fix Operator image repo/tag variable usage

### PR DESCRIPTION
**Pull request by erichsueh3**
**Description:** <Describe what has changed.>
This PR fixes how Operator image repo/tag variables are set so that they can be properly overridden with a `TF_VAR_` env variable export in the Operator CI, see [here](https://github.com/aws-observability/aws-otel-collector/blob/main/.github/workflows/CI-Operator.yml#L173). 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: terraform/eks/adot-operator/adot_operator.tf [hunk 1]
```diff
@@ -24,11 +24,9 @@ variable "kubeconfig" {
 }
 
 variable "operator_repository" {
-  default = "ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator"
 }
 
 variable "operator_tag" {
-  default = "latest"
```
Comment (bryan-aguilar) [inline terraform/eks/adot-operator/adot_operator.tf:31]: Can we add a type to these variables?

**Review by bryan-aguilar — APPROVED** [review]

File: terraform/eks/adot-operator/adot_operator.tf [hunk 2]
```diff
@@ -24,11 +24,9 @@ variable "kubeconfig" {
 }
 
 variable "operator_repository" {
-  default = "ghcr.io/open-telemetry/opentelemetry-operator/opentelemetry-operator"
 }
 
 variable "operator_tag" {
```
Comment (bryan-aguilar) [inline terraform/eks/adot-operator/adot_operator.tf:29]:
```suggestion
variable "operator_tag" {
  type = string
```

File: terraform/eks/adot-operator/adot_operator.tf [hunk 3]
```diff
@@ -24,11 +24,9 @@ variable "kubeconfig" {
 }
 
 variable "operator_repository" {
```
Comment (bryan-aguilar) [inline terraform/eks/adot-operator/adot_operator.tf:26]:
```suggestion
variable "operator_repository" {
  type = string
```

**Comment by erichsueh3** [comment]
@bryan-aguilar added type to the variables, but in `variables.tf` rather than `adot_operator.tf` because it's more readable logically IMO

**Review by vasireddy99 — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]
Approved and merged, can you also make a PR to ensure this ends up in the `release/v0.21.x` branch.

**Pull request merged**